src/jobservice/job/impl/gc/garbage_collection.go

```diff
@@ -350,24 +357,14 @@ func (gc *GarbageCollector) cleanCache() error {
 // deletedArt contains the two parts of artifact
 // 1, required part, the artifacts were removed from Harbor.
 // 2, optional part, the untagged artifacts.
-func (gc *GarbageCollector) deletedArt(ctx job.Context) (map[string][]string, error) {
+func (gc *GarbageCollector) deletedArt(ctx job.Context) (map[string][]model.ArtifactTrash, error) {
```

Comment:
`map[string][]model.ArtifactTrash` => `map[string][]*model.ArtifactTrash`, otherwise for every loop of []model.ArtifactTrash， struct copy will be happend.